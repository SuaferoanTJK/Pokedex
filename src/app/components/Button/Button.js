import React from "react";
import PropTypes from "prop-types";

function Button(props) {
  const { textButton, char, addPokemons, removePokemons } = props;

  return (
    <button
      onClick={() => {
        textButton === "Add" ? addPokemons(char) : removePokemons(char);
      }}
      className="Button"
    >
      {textButton}
    </button>
  );
}

export default Button;

Button.propTypes = {
  textButton: PropTypes.string,
};
