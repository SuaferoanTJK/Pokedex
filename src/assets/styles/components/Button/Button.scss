.Button {
  cursor: pointer;
  border: none;
  padding: 0.5rem 1rem;
  margin-bottom: 1rem;
  border-radius: 1rem;
  &:hover {
    background-color: #000;
    border: 0.15rem solid #fff;
    color: #fff;
    font-weight: bold;
  }
}
