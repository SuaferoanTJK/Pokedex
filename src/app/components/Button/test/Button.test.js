import React from "react";
import "@testing-library/jest-dom";
import { mount, shallow } from "enzyme";
import Button from "../Button";

describe("unit tests Button", () => {
  it("It should display the component correctly <Button />", () => {
    const wrapper = shallow(
      <Button
        textButton={"ADD"}
        char={{ id: 1, name: "bulbasaur" }}
        addPokemons={() => {}}
        removePokemons={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  // it("It should render 'Add' if prop.textButton is 'Add'  ", () => {
  //   const wrapper = shallow(
  //     <Button
  //       textButton="Add"
  //       char={{ id: 1, name: "bulbasaur" }}
  //       addPokemons={() => {}}
  //       removePokemons={() => {}}
  //     />
  //   );
  //   expect(wrapper.find("button").at(0).text()).toEqual("Add");
  // });
});
