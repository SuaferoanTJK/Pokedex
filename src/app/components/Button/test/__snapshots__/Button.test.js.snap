// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests Button It should display the component correctly <Button /> 1`] = `
<button
  className="Button"
  onClick={[Function]}
>
  ADD
</button>
`;
